textproc/libxslt/Portfile

```diff
@@ -34,10 +34,11 @@ configure.args  --without-python \
                 --disable-silent-rules
 
 if {${name} eq ${subport}} {
-    revision            1
+    revision            2
 
     depends_build       port:pkgconfig
     depends_lib-append  port:libiconv \
+                        port:icu \
```

Comment:
This is a case of overlinking, and appears to be happening because we build static libraries. The configure script is aware of xml-config's --dynamic option, but doesn't use it unless only dynamic libraries are being built. When both are built, it uses the static ldflags then linking the dynamic libs too.

Reply:
What do you suggest to do here? Don't build the static libraries? Or change this to always use `--dynamic` (will that work)?

```shell
        if test "$build_static_libs" = "no"; then
            LIBXML_LIBS="-L${LIBXML_SRC} `$XML_CONFIG --libs --dynamic`"
        else
            LIBXML_LIBS="-L${LIBXML_SRC} `$XML_CONFIG --libs`"
        fi
```

Reply:
Neither. The build system needs to be redesigned to use the output of `xml-config --libs --dynamic` when building the dynamic library and the output of `xml-config --libs` when building the static library. I feel like I've discussed this problem with them before...

Reply:
Looking through `$prefix/lib`, it doesn't appear that all the ports install a static library. From a quick search, `libassuan`, `libgcrypt`, `libgpg-error`, `libksba`, `libLASi`, `libmagic`, `libmodplug`, `libomp` only install a dynamic library. Can this be done for `libxslt`.

Reply:
You are correct that not all ports install static libraries. Some use build systems that force you to choose either static or dynamic libraries and in those cases certainly we choose the dynamic libraries. CMake is an example of such a build system. But autotools typically allows both to be built simultaneously and it has up to now been our suggestion that ports should install both static and dynamic libraries if that's possible.

Reply:
> The build system needs to be redesigned to use the output of `xml-config --libs --dynamic` when building the dynamic library and the output of `xml-config --libs` when building the static library. I feel like I've discussed this problem with them before...

Here's the bug report: https://gitlab.gnome.org/GNOME/libxslt/-/issues/36

They don't know how to fix it and welcome any help.